### PR TITLE
DMP-2245 - move validate_only to query param

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-2245

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/retention/controller/RetentionController.java [hunk 1]
```diff
@@ -43,9 +43,10 @@ public ResponseEntity<List<GetCaseRetentionsResponse>> retentionsGet(Integer cas
             securityRoles = {JUDGE, REQUESTER, APPROVER, TRANSCRIBER, TRANSLATION_QA, ADMIN},
             globalAccessSecurityRoles = {JUDGE, ADMIN})
     @Override
-    public ResponseEntity<PostRetentionResponse> retentionsPost(PostRetentionRequest postRetentionRequest) {
+    public ResponseEntity<PostRetentionResponse> retentionsPost(Boolean validateOnly,
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/darts/retention/controller/RetentionController.java:46]: minor: shall we add `@RequestParam` ? just to be consistent with what we do in other controllers

File: src/test/java/uk/gov/hmcts/darts/retention/service/impl/RetentionPostServiceImplTest.java [hunk 2]
```diff
@@ -130,7 +130,7 @@ void fail_NoCase() {
 
         var exception = assertThrows(
```
Comment (mario-paniccia) [inline src/test/java/uk/gov/hmcts/darts/retention/service/impl/RetentionPostServiceImplTest.java:131]: shall we add a test where we verify that if the query param is null it defaults to `false` as per the specification?

**Review by mario-paniccia — APPROVED** [review]
just a couple of minor comments if you want

File: src/main/java/uk/gov/hmcts/darts/retention/controller/RetentionController.java (hunk 1 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/retention/controller/RetentionController.java:46]: there was actually a comment one time to not do that, as it appears in the RetentionApi that we are implementing (auto generated in there) and so we were asked not to duplicate it in the implementation.

File: src/test/java/uk/gov/hmcts/darts/retention/service/impl/RetentionPostServiceImplTest.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/test/java/uk/gov/hmcts/darts/retention/service/impl/RetentionPostServiceImplTest.java:131]: scenario now added
